test/unit/cmap/connection.test.ts

```diff
@@ -333,5 +336,49 @@ describe('new Connection()', function () {
       expect(stream.read(1)).to.deep.equal(Buffer.from([6, 0, 0, 0, 5, 6]));
       expect(stream.read(1)).to.equal(null);
     });
+
+    it('parses many wire messages when chunk arrives', function () {
```

Comment:
can we add a test for several full messages + partial so that we can ensure we are still behaving correctly in those cases?

Reply:
Added a many messages and partial case and asserted the sizes of the chunks just to be sure.